classes/Pods.php

```diff
@@ -1015,12 +1015,25 @@ public function field( $name, $single = null, $raw = false ) {
 						$field_names = explode( '.', $params->name );
 
 						if ( isset( $field_names[1] ) ) {
-							$size = $field_names[1];
+							$size    = $field_names[1];
+							$sizes   = get_intermediate_image_sizes();
+							// Not shown by default.
+							$sizes[] = 'full';
+							$sizes[] = 'original';
+							if ( ! in_array( $size, $sizes, true ) ) {
+								// No valid image size found.
+								$size = false;
+							}
 						}
 					}
 
-					// Pods will auto-get the thumbnail ID if this isn't an attachment.
-					$value = pods_image( $this->id(), $size, 0, null, true );
+					if ( $size ) {
+						// Pods will auto-get the thumbnail ID if this isn't an attachment.
+						$value = pods_image( $this->id(), $size, 0, null, true );
+					} else {
+						array_shift( $field_names );
```

Comment:
would nee to test that but from skimming over the code (never really been in here) but shouldn't just 
$object_field_found = false; 
be used and let the rest be done by the traversal code? guess that might need changes down the line but… it's difficult as pods has no idea of the "relationship" I guess - does that makes sense?

Reply:
Since it's about the WP core metadata for thumbnails it's not an object field, but also not a Pods field.
See my initial comment on this PR. It would indeed be nice to include the post thumb as a Pods field by default so all file-field related functionality is possible with the post thumbnail.